src/index.test.ts

```diff
@@ -120,6 +120,24 @@ describe('jsonFetch', function () {
       expect(response.statusText).to.equal('Created');
       expect(response.headers).to.be.ok();
     });
+
+    it('calls onRequestStart when is passed', async function () {
+      const onRequestStart = sandbox.spy(() => {});
+      nock('http://www.test.com').get('/products/1234').reply(200);
+      await jsonFetch('http://www.test.com/products/1234', {
+        onRequestStart,
+      });
+      expect(onRequestStart.callCount).to.equal(1);
```

Comment:
Add assertion on what was passed to it? It's pretty critical so I think it's worth testing.

Reply:
I'm not sure how to do that with the `sandbox.stub()` call, could you throw a suggestion here please?

Reply:
```js
expect(onRequestStart).to.have.been.calledWithMatch({
  url: 'http://www.test.com/products/1234',
  retryCount: 1,
  headers: {/* ...  */},
  // etc.?
});
```

We can pair if this isn't working.